libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -389,6 +393,45 @@ export class CartService {
     await this.cartManager.updateFreshCart(cartId, version, cartDetails);
   }
 
+  /**
+   * Fetch the invoice preview for the latest invoice associated with a cart
+   */
+  async previewLatestInvoice(
+    cartId: string
+  ): Promise<InvoicePreview | undefined> {
+    const cart = await this.cartManager.fetchCartById(cartId);
+    if (!cart.stripeSubscriptionId) return;
+
+    const subscription = await this.subscriptionManager.retrieve(
+      cart.stripeSubscriptionId
+    );
+    const paymentIntent = await this.subscriptionManager.getLatestPaymentIntent(
+      subscription
+    );
+    if (!paymentIntent || !paymentIntent.invoice) return;
+    const latestInvoice = await this.invoiceManager.retrieve(
+      paymentIntent.invoice
+    );
+    const latestInvoiceDTO = stripeInvoiceToInvoicePreviewDTO(latestInvoice);
+
+    const paymentMethodId =
+      typeof paymentIntent.payment_method === 'string'
+        ? paymentIntent.payment_method
+        : paymentIntent.payment_method?.id;
+
+    if (!paymentMethodId) return latestInvoiceDTO;
+
+    const paymentMethod = await this.stripeClient.paymentMethodRetrieve(
+      paymentMethodId
+    );
+    const last4 = paymentMethod.card?.last4;
```

Comment:
Not sure this should be considered as part of the invoice preview

Reply:
It should exist somewhere between the `invoice` layer, and the `cart` layer. It seemed like `invoicePreview` was effectively an aggregate of fields related to invoices, rather than a method for surfacing the actual invoice itself.

In my brain, the specifics of payment details exist in relation to the payment event (here being on the completed invoice). Do you have a suggestion for where this should live?